File: src/ProjectR.Scripting/content/scripts/spells/CripplingAura.lua
function GetTargetType()
    return Enemies
end

function GetName()
  return "Crippling Aura"
end

function GetDescription()
  return "Focus your aura to cripple your enemies.\nReduces AD, MD, DEF & MR of enemies."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .88
end

function GetDelay()
  return 0
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  d:BuffStat(AD, -1)
  d:BuffStat(MD, -1)
  d:BuffStat(DEF, -1)
  d:BuffStat(MR, -1)
end

File: src/ProjectR.Scripting/content/scripts/spells/CripplingAura.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class CripplingAura : SpellScriptBase
    {
        public override string Name { get { return "Crippling Aura"; } }
        public override string Description { get { return "Focus your aura to cripple your enemies.\nReduces AD, MD, DEF & MR of enemies."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 88; } }
        public override double Delay { get { return 0; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            BuffStat(Stat.AD, -1);
            BuffStat(Stat.MD, -1);
            BuffStat(Stat.DEF, -1);
            BuffStat(Stat.MR, -1);
        }
    }
}